File: Data Structures & Algorithms/find-duplicate-integer/submission-3.java
class Solution {
    public int findDuplicate(int[] nums) {
        int[] seen = new int[nums.length + 1];
        for (int n : nums) {
            seen[n]++;
        }

        for (int i = 0; i < seen.length; i++) {
            if (seen[i] > 1) {
                return i;
            }
        }

        return -1;
    }
}

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-3.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        arr = []
        l = 0
        
        while l < len(nums):
            product = 1
            for i in range(len(nums)):
                if i == l: 
                    continue
                
                product = product * nums[i]
            arr.append(product)
            l += 1
        
        return arr

